semgrep/semgrep/rule_match.py

```diff
@@ -125,6 +125,10 @@ def lines(self) -> List[str]:
         start_line = self.start["line"] - 1
         end_line = self.end["line"]
 
+        if start_line == -1 and end_line == 0:
```

Comment:
This smells off to me. Good defensive programming I guess, but semgrep shouldn't return a match here.

Reply:
Should .* not match an empty file?

Reply:
> ... but semgrep shouldn't return a match here.

IMO `pattern-regex` is a bit of an outlier, and it seems reasonable for it to return an empty match:

```
In [10]: print(re.search("^.*", ""))                                                                    
<_sre.SRE_Match object; span=(0, 0), match=''>

In [11]: print(re.search("^.+", ""))                                                                    
None
```

This type of rule is also important for our file detection rules (https://github.com/returntocorp/semgrep/issues/2438#issuecomment-891039714). In this case we don't really care if the file is empty or not, only if it exists. So I would vote that we do want to support empty matches, if only for this specific use-case.